packages/primitives-core/src/types.ts

```diff
@@ -1,10 +1,24 @@
-type NamedStyles<T> = { [P in keyof T]: unknown }
+// This is based on the StyleSheet types from @types/react-native
+
+type Falsy = undefined | null | false
+
+interface RecursiveArray<T>
+  extends Array<T | ReadonlyArray<T> | RecursiveArray<T>> {}
+
+type RegisteredStyle<T> = number & { __registeredStyleBrand: T }
+
+type StyleProp<T> =
+  | T
+  | RegisteredStyle<T>
+  | RecursiveArray<T | RegisteredStyle<T> | Falsy>
+  | Falsy
+
+type NamedStyles<T> = { [P in keyof T]: any }
 
-// This is based on the StyleSheet type from @types/react-native
 export interface AbstractStyleSheet {
   create<T extends NamedStyles<T> | NamedStyles<any>>(
     styles: T | NamedStyles<T>
-  ): T
+  ): { [P in keyof T]: number }
```

Comment:
`@types/react-native` has the return type as `T` which seems wrong to me, and it was causing issues... This is the thing I am not totally sure is correct.